### PR TITLE
[BACKLOG-6092] Prompt API - Add a state API to read and modify the st…

### PR DESCRIPTION
**Pull request by annapopova1**
…ate of the prompting system

* implemented api.operation.state API function
* included read only properties validation
* included types validation
* included setting page validation
* wrote tests

@diogofscmariano @krivera-pentaho @pamval @plagoa @jvelasques @afrjorge please review

### CODE REVIEW

**Comment by diogofscmariano** [comment]
@annapopova1 can you please make sure a subtask for Virginia is opened? 
merging

File: package-res/resources/web/prompting/PromptPanel.js [hunk 1]
```diff
@@ -33,6 +33,29 @@
  */
 define(['cdf/lib/Base', 'cdf/Logger', 'dojo/number', 'dojo/i18n', 'common-ui/util/util', 'common-ui/util/GUIDHelper', './WidgetBuilder', 'cdf/Dashboard.Clean', './parameters/ParameterDefinitionDiffer', 'common-ui/jquery-clean'],
     function (Base, Logger, DojoNumber, i18n, Utils, GUIDHelper, WidgetBuilder, Dashboard, ParamDiff, $) {
+
+      var _STATE_CONSTANTS = {
+        readOnlyProperties: ["promptNeeded", "paginate", "totalPages", "showParameterUI", "allowAutoSubmit"],
+        msgs: {
+          notChangeReadonlyProp: function(readOnlyProperties) {
+            return "Not possible to change the read only properties: " + readOnlyProperties;
```
Comment (diogofscmariano) [inline package-res/resources/web/prompting/PromptPanel.js]: @annapopova1 please add these messages for Virginia review

**Comment by annapopova1** [comment]
@diogofscmariano I will prepare sub task. Thank you.

File: package-res/resources/web/prompting/PromptPanel.js [hunk 2]
```diff
@@ -1391,6 +1493,87 @@ define(['cdf/lib/Base', 'cdf/Logger', 'dojo/number', 'dojo/i18n', 'common-ui/uti
...
+         *   "autoSubmit":true,
+         *   "page":5
+         * };
+         *
+         * var updatedState = api.operation.state(state);
+         * // Return value:
+         * //   {
+         * //     "promptNeeded":false,
+         * //     "paginate":true,
+         * //     "totalPages":1,
```
Comment (diogofscmariano) [inline package-res/resources/web/prompting/PromptPanel.js]: @annapopova1 total pages 1 and page 5? :P 
can you please fix this?

Comment (annapopova1) [inline package-res/resources/web/prompting/PromptPanel.js]: I will fix together with doc changes after review.

File: package-res/resources/web/prompting/PromptPanel.js [hunk 3]
```diff
@@ -1391,6 +1493,87 @@ define(['cdf/lib/Base', 'cdf/Logger', 'dojo/number', 'dojo/i18n', 'common-ui/uti
...
+          };
+          return result;
+        },
+
+        /**
+         * Modifys a state of the prompting system.
+         *
+         * @name PromptPanel@setState
+         * @method
+         * @param {Object} [state]                    The set of flags which will be applyed to current state. It's optional parameter.
```
Comment (diogofscmariano) [inline package-res/resources/web/prompting/PromptPanel.js]: @annapopova1  is the stage really optional? does it makes sense to call setState without the argument

Comment (annapopova1) [inline package-res/resources/web/prompting/PromptPanel.js]: My mistake. I will fix with doc.

**Pull request merged**